tool/tctl/common/autoupdate_command.go

```diff
@@ -0,0 +1,206 @@
+/*
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package common
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/alecthomas/kingpin/v2"
+	"github.com/coreos/go-semver/semver"
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport"
+	"github.com/gravitational/teleport/api/client/webclient"
+	"github.com/gravitational/teleport/api/gen/proto/go/teleport/autoupdate/v1"
+	update "github.com/gravitational/teleport/api/types/autoupdate"
+	"github.com/gravitational/teleport/lib/auth/authclient"
+	"github.com/gravitational/teleport/lib/service/servicecfg"
+	"github.com/gravitational/teleport/lib/utils"
+	"github.com/gravitational/teleport/lib/utils/interval"
+)
+
+// versionResponse is structure for formatting the autoupdate version response.
+type versionResponse struct {
+	ToolsVersion string `json:"tools_version"`
+}
+
+// AutoupdateCommand implements the `tctl autoupdate` command for managing
+// autoupdate process for tools and agents.
+type AutoupdateCommand struct {
+	app    *kingpin.Application
+	config *servicecfg.Config
+
+	updateCmd *kingpin.CmdClause
+	getCmd    *kingpin.CmdClause
+	watchCmd  *kingpin.CmdClause
+
+	toolsAutoupdate        string
+	toolsAutoupdateVersion string
+	proxy                  string
+}
+
+// Initialize allows AutoupdateCommand to plug itself into the CLI parser.
+func (c *AutoupdateCommand) Initialize(app *kingpin.Application, config *servicecfg.Config) {
+	c.app = app
+	c.config = config
+	autoupdateCmd := app.Command("autoupdate", "Teleport autoupdate commands.")
+
+	c.updateCmd = autoupdateCmd.Command("update", "Edit autoupdate configuration.")
+	c.updateCmd.Flag("set-tools-auto-update", `Enable or disable tools autoupdate in cluster.`).EnumVar(&c.toolsAutoupdate, "on", "off")
```

Comment:
```suggestion
	c.updateCmd.Flag("set-tools-autoupdate", `Enable or disable tools autoupdate in cluster.`).EnumVar(&c.toolsAutoupdate, "on", "off")
```
Feel free to ignore, but if we want to be super-consistent about it, could change

Reply:
missed this one, will update with rfd